src/main/scala/com/typesafe/sbt/packager/distroless/DistrolessPlugin.scala

```diff
@@ -0,0 +1,262 @@
+  }
+
+  /**
+    * @param dockerBaseDirectory
+    *   the installation directory
+    * @return
+    *   COPY command copying all files inside the installation directory
+    */
+  private final def makeCopyLayerDirect(layerId: Option[Int], dockerBaseDirectory: String): CmdLike = {
+
+    /**
+      * This is the file path of the file in the Docker image, and does not depend on the OS where the image is being
+      * built. This means that it needs to be the Unix file separator even when the image is built on e.g. Windows
+      * systems.
+      */
+    val files = dockerBaseDirectory.split(UnixSeparatorChar)(1)
+    val path = layerId.map(i => s"$i/$files").getOrElse(s"$files")
+    Cmd("COPY", s"$path /$files")
+  }
+
+  /**
+    * @param exposedPorts
+    * @return
+    *   if ports are exposed the EXPOSE command
+    */
+  private final def makeExposePorts(
+    exposedPorts: Seq[Int],
+    exposedUdpPorts: Seq[Int],
+    jdwpPort: Option[Int]
+  ): Option[CmdLike] =
+    if (exposedPorts.isEmpty && exposedUdpPorts.isEmpty && jdwpPort.isEmpty) None
+    else
+      Some(
+        Cmd(
+          "EXPOSE",
+          (exposedPorts.map(_.toString) ++ exposedUdpPorts.map(_.toString).map(_ + "/udp") ++ jdwpPort
+            .map(_.toString)
+            .toSeq) mkString " "
+        )
+      )
+
+  /**
+    * @param entrypoint
+    * @return
+    *   ENTRYPOINT command
+    */
+  private final def makeEntrypoint(entrypoint: Seq[String]): CmdLike =
+    ExecCmd("ENTRYPOINT", entrypoint*)
+
+  /**
+    * Default CMD implementation as default parameters to ENTRYPOINT.
+    * @param args
+    * @return
+    *   CMD with args in exec form
+    */
+  private final def makeCmd(args: Seq[String]): CmdLike =
+    ExecCmd("CMD", args*)
+
+  /**
+    * Builds the classpath argument to use when running java. This works by concatenating all dependencies in the lib
+    * folder with a colon
+    * @param baseDirectory
+    *   where is the base docker directory (ie. /opt/docker)
+    * @param dependencies
+    *   list of dependencies as JAR files
+    * @return
+    *   The <pre>-cp</pre> argument with all dependencies
+    */
+  private def makeClasspath(baseDirectory: String, dependencies: Seq[String]): Seq[String] = {
+    val dependenciesBaseDirectory = baseDirectory + "/lib/"
+
+    Seq("-cp", dependencies.map(dependenciesBaseDirectory + _).mkString(":"))
+  }
+
+  /**
+    * Builds the JDWP cli argument to be used when running java
+    * @param port
+    *   port to expose the remote debugger on
+    * @return
+    */
+  private def makeJdwpDebugger(port: Int): String =
+    s"-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=$port"
+
+  /**
+    * Tries to define what main class to call when running the distroless container
+    * @param mainClass
+    *   optional main class
+    * @param discoveredMainClasses
+    *   discovered main classes
+    * @return
+    */
+  private def defineMainClass(
```

Comment:
This method is similar to `CommonStartScriptGenerator.generateStartScripts` but here we require a single main class as it has to be defined at build time. The user cannot specify it at runtime like it was possible with the script.